users.py

```diff
@@ -45,52 +50,54 @@ def create_user():
         # psycopg2 sql injection
         cursor, conn = get_db_connection()
         if cursor is None or conn is None:
-            return jsonify({'error' : 'Failed to connect to server' }), 500
+            return jsonify({'error' : 'Internal Error' }), 500
         
-        # Checking if a user with this email already exists
-        cursor.execute("SELECT id FROM users WHERE email = %s", (email,))
-        existing_user = cursor.fetchone()
-        if existing_user:
-            cursor.close()
-            conn.close()
-            return jsonify({'error' : 'User with this email already exist!'}), 409
-
         cursor.execute("""
             INSERT INTO users (id, first_name, last_name, email, password_hash, created_at, updated_at) 
             VALUES (%s, %s, %s, %s, %s, %s, %s) RETURNING id;
         """, (user_id, data['first_name'], data['last_name'], email, hashed_password, str(created_at), str(updated_at)))
+        
         conn.commit() 
         cursor.close()
         conn.close()
-        # more error hanlding regarding - like "user already exist"
+
     except Exception as e:
-        return jsonify({'error': 'Error creating a new user'}), 500
+        logging.debug("Error creating user: ", exc_info=True)
+        return jsonify({'error': 'Internal Error'}), 500
+    
+    except UniqueViolation as UV:
+        logging.debug(f"Unique constraint violation for email {email}: {UV}")
+        return jsonify({'error': 'Try another email.'}), 409
 
     # Return the created user data
     return jsonify({'id': user_id, 'first_name': data['first_name'], 'last_name': data['last_name'], 'email': data['email']}), 201
 
-# Get All Users
-@users_bp.route('/users', methods=['GET'])
-def get_users():
-    # Retrieve users from the database (handle database exceptions)
+# Get User
+@users_bp.route('/get_user', methods=['GET']) 
+def get_user():
+    #data = request.get_json() 
+    email = request.args.get('email') 
+
+    if not email:
+        logging.error("Email not provided", exc_info=True)
+        return jsonify({'error': "'email' is required."}), 400
+
     try:
-        cursor, conn = get_db_connection()
-        if cursor is None or conn is None:
-            return jsonify({'Error' : 'Failed to connect to server' }), 500
+        user = get_user_by_email(email)  
```

Comment:
GET /users should return the user scoped down by the user_id from the auth token provided. In the future, we will expand it to also "admin" users that can see everything.